frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+const { utils } = require( "ethers" );
+// import utils from "ethers";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{utils.formatEther(data.balance)}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		const provider = await detectEthereumProvider();
+		if (!provider) {
+			console.error("no ethereum provider");
```

Comment:
For the design, will we need a user-facing error message later?

Reply:
let us decide it in later phase of the project

Reply:
So, when would this be changed? I think, this should be done now because later we might forget about this and this is important for usability.